server/http/helloapp/install.go

```diff
@@ -39,21 +52,106 @@ func (h *helloapp) handleInstall(w http.ResponseWriter, req *http.Request, claim
 	return http.StatusOK, nil
 }
 
-func (h *helloapp) handleConnectedInstall(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, data *apps.CallData) (int, error) {
+func (h *helloapp) handleConnectedInstall(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, data *apps.CallRequest) (int, error) {
 	var teams []*model.Team
+	var team *model.Team
+	var channel *model.Channel
+
 	err := h.asUser(data.Context.ActingUserID,
-		func(client *model.Client4) error {
-			teams, _ = client.GetAllTeams("", 0, 100)
+		func(mmclient *model.Client4) error {
+			var api4Resp *model.Response
+			teams, api4Resp = mmclient.GetAllTeams("", 0, 1)
+			if api4Resp.Error != nil {
+				return api4Resp.Error
+			}
+			if len(teams) == 0 {
+				return errors.New("no team found to create the Hallo სამყარო channel")
+			}
+
+			// TODO call a Modal to select a team
+			team = teams[0]
+
+			// Ensure "Hallo სამყარო" channel
+			channel, _ = mmclient.GetChannelByName(AppID, team.Id, "")
+			if channel != nil {
+				// TODO DM to user that the channel has been found
+				if channel.DeleteAt != 0 {
+					return errors.Errorf("TODO unarchive channel %s \n", channel.DisplayName)
+				}
+				h.DM(data.Context.ActingUserID, "Found eisting ~%s channel.", AppID)
+			} else {
+				channel, api4Resp = mmclient.CreateChannel(&model.Channel{
+					TeamId:      team.Id,
+					Type:        model.CHANNEL_OPEN,
+					DisplayName: AppDisplayName,
+					Name:        AppID,
+					Header:      "TODO header",
+					Purpose:     `to say, "Hallo სამყარო!"`,
+				})
+				if api4Resp.Error != nil {
+					return api4Resp.Error
+				}
+
+				h.DM(data.Context.ActingUserID, "Created ~%s channel.", AppID)
+			}
+
+			// Add the Bot user to the team and the channel.
+			_, api4Resp = mmclient.AddTeamMember(team.Id, data.Context.App.BotUserID)
+			if api4Resp.Error != nil {
+				return api4Resp.Error
+			}
+			_, api4Resp = mmclient.AddChannelMember(channel.Id, data.Context.App.BotUserID)
```

Comment:
Do bots need to be channel members? I feel like that crowds the channel members list if that's the case.

Reply:
Bot (accounts/access tokens) have to be added to the Teams and Channels to have access to them.